packages/allow-scripts/src/setup.js

```diff
@@ -17,36 +17,55 @@ function addInstallParentDir(filename) {
   return path.join(rootDir, filename)
 }
 
+function writeRcFileContent({file, exists, entry}){
+  let rcPath = addInstallParentDir(file)
+
+  if (!exists) {
+    writeFileSync(rcPath, entry + '\n')
+    console.log(`@lavamoat/allow-scripts: created ${rcPath} file with entry: ${entry}.`)
+    return
+  }
+
+  const rcFileContents = readFileSync(rcPath, 'utf8')
+  if (rcFileContents.includes(entry)) {
+    console.log(`@lavamoat/allow-scripts: file ${rcPath} already exists with entry: ${entry}.`)
+  } else {
+    appendFileSync(rcPath, entry + '\n')
+    console.log(`@lavamoat/allow-scripts: added entry to ${rcPath}: ${entry}.`)
+  }
+}
+
 function writeRcFile () {
   const yarnRcExists = existsSync(addInstallParentDir('.yarnrc'))
+  const yarnYmlExists = existsSync(addInstallParentDir('.yarnrc.yml'))
   const npmRcExists = existsSync(addInstallParentDir('.npmrc'))
   const yarnLockExists = existsSync(addInstallParentDir('yarn.lock'))
 
-  let rcFile, rcEntry
+  const configs = []
   if (yarnRcExists || yarnLockExists) {
-    rcFile = '.yarnrc'
-    rcEntry = 'ignore-scripts true'
-  } else {
-    // if no lockfiles exixt, default to npm, because that's what everyone has anyway
-    rcFile = '.npmrc'
-    rcEntry = 'ignore-scripts=true'
+    configs.push({
+      file: ".yarnrc",
+      exists: yarnRcExists,
+      entry: "ignore-scripts true",
+    })
   }
-
-  let rcPath = addInstallParentDir(rcFile)
-
-  if (!yarnRcExists && !npmRcExists) {
-    writeFileSync(rcPath, rcEntry + '\n')
-    console.log(`@lavamoat/allow-scripts: created ${rcPath} file with entry: ${rcEntry}.`)
-    return
+  if (yarnYmlExists || yarnLockExists) {
```

Comment:
I decided to create both files if yarn.lock exists, because people may run any version of yarn and that's going to remain a point of confusion for years.
More advanced users can remove the excess file if they decide to do so.